### DISCUSSION TITLE
Problem with files path after buidl

### DISCUSSION THREAD

**Discussion by Evanshion** [marcelotduarte/cx_Freeze#1776] [Q&A] [answered]
Hi all, I have a problem when I launch my application, I see that it tries to pick resource from this path:
![image](https://user-images.githubusercontent.com/117465084/217276912-56a19e0c-4217-473e-b292-f97487c61f68.png)
(Forgive me for the screenshot)
I the right path should be the app.exe's folder/Languages.
I can't understand why it goes on lib/Library.zip/Languages....ecc..
Can you help me? 
This is my setup
```import sys
import os
import cx_Freeze 


executables = [cx_Freeze.Executable("voicechecker.py", icon="Cefla.ico")]

include_files = ["img/", "Languages/", "Reports/", "SAM_SOCKET/", "Voices/", "Cefla.ico"]
path = os.path.dirname(os.path.abspath(__file__))
cx_Freeze.setup(
    name = "Cefla Voice Assistant",
    options = {
    "build_exe": {
        "include_files": include_files,
        "path": path
    }},
    author = "Giuseppe Ieno",
    version = "1.0",
    description = "Tool for automated testing for Cefla vocal commands (Vivoka)",
    executables = executables
)
```

**Comment by marcelotduarte** [comment, marked as answer]
> I the right path should be the app.exe's folder/Languages. I can't understand why it goes on lib/Library.zip/Languages....ecc.. 

This is defined in the app? (The usage of 'path' in the setup does not change this and should be removed).
Are you using some external package to load this file?
Please read this: https://cx-freeze.readthedocs.io/en/latest/faq.html#using-data-files

**Reply by marcelotduarte** [reply]
In setup you use include_files to copy the data files. In your code you should use:
```python
if getattr(sys, "frozen", False):
    # The application is frozen
    ROOT_DIR = os.path.dirname(sys.executable)
else:
    # The application is not frozen
    # Change this bit to match where you store your data files:
    ROOT_DIR = os.path.dirname(__file__)
```

**Comment by Evanshion** [comment]
Ok, I'm trying to follow your suggestion and I removed "path".
Actually I call my files, in my application, in this way:
```ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
#ROOT_DIR = os.path.dirname(sys.executable)
REPORT_DIR = ROOT_DIR+'\\Reports'
VOICE_DIR = ROOT_DIR+'\\Voices'
LAN_DIR = ROOT_DIR+'\\Languages'
RELATIVE_DIR =ROOT_DIR+'\\SAM_SOCKET'```

So if I have to call, for example, an excel file, i can write something like this (if the file, is in LAN_DIR):
`path = ROOT_DIR+'\\Reports\\'+filename`
and it works well. (when I use QTCreator)

I saw the function you suggested but I didn't understand how and where I should use it:

1. in my code? 
2. in my setup.py?
And for both question, in which way?
I tried to looking for some examples but I failed.
Thanks in advance for your support.

**Reply by Evanshion** [reply]
It worked, thank you a lot
